feat(install): Download and launch UI Builder (#2720)
bonitasoft/bonita-doc: modules/applications/pages/download-and-launch.adoc

= Download and launch Bonita UI Builder
:page-aliases: ROOT:download-and-launch.adoc
:description: Describes how to download and launch Bonita UI Builder.

[NOTE]
====
For Subscription editions only.
====

== Create the docker environment

- Create a `docker-compose.yml` file with the content below:

(_To update with the final content_)

.Click to see the docker compose file
[%collapsible]
====
[source, shell]
----
services:

>   bonitappsmith:

Reviewer: ```suggestion
	bonita-ui-builder:
```

>     image: bonitasoft.jfrog.io/docker/bonitappsmith:0.0.1

Reviewer: ```suggestion
    image: bonitasoft.jfrog.io/docker/bonita-ui-builder:0.0.1
```

>     container_name: bonitappsmith

Reviewer: ```suggestion
    container_name: bonita-ui-builder
```

            proxy_set_header X-Real-IP $remote_addr;  # Forwards the real IP address.
            proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;  # Forwards the X-Forwarded-For header.
            proxy_set_header X-Forwarded-Proto $scheme;  # Forwards the request scheme.
            proxy_http_version 1.1;  # Sets the HTTP version used in the proxy request.
            proxy_set_header Accept-Encoding "";  # Clears the Accept-Encoding header.
            proxy_set_header X-Forwarded-Host $host;  # Forwards the original host requested.
            proxy_set_header X-Forwarded-Port $server_port;  # Forwards the server port.
            proxy_set_header Upgrade $http_upgrade;  # Handles protocols upgrades, e.g., for WebSocket.
            proxy_set_header Connection "upgrade";  # Maintains upgrade header connections.
        }

        # Named location for handling redirections when unauthorized access is detected.
        location @redirect_to_login {
            return 302 http://localhost/bonita/apps/sendRedirect?redirectUrl=$encoded_uri;  # Redirects to the login page, appending the original requested URL.
        }

>     }

Reviewer: Can we imagine a way to give this nginx configuration more easier, and more maintainable than documentation content?

Author: Yes, I think we'll have to deliver the docker-compose and nginx files at some point. In the meantime, I think we have to provide them here...

====

== Download and launch the docker image

> The UI Builder docker image is located in the xref:software-extensibility:bonita-repository-access.adoc[Bonita Artifact Repository].

Reviewer 2: ```suggestion
Bonita UI Builder docker image is located in the xref:software-extensibility:bonita-repository-access.adoc[Bonita Artifact Repository].
```

Once you got the credentials, you have to authenticate with the `docker login` command:
[source, shell, subs="+attributes"]
----
docker login {bonitasoft-registry}
----

> Run the following command to launch the Bonita UI Builder:

Reviewer 2: ```suggestion
Run the following command to launch Bonita UI Builder:
```

[source, shell]
----
docker compose up -d
----

> Then, you can access the Bonita UI Builder at `http://localhost`.

Reviewer 2: ```suggestion
Then, you can access Bonita UI Builder at `http://localhost`.
```

> Then, you can access the Bonita UI Builder at `http://localhost:8090`.

Reviewer: It would become this url soon
```suggestion
Then, you can access the Bonita UI Builder at `http://localhost`.
```

====
The first time, the initialisation can take a while, a waiting message appears. Reload the page if the page is blank.
====

== Start Bonita on the local machine

> To use the Bonita UI Builder, a Bonita runtime running on your local machine is required (it may be a Bonita Studio or a standalone Bonita runtime).

Reviewer 2: ```suggestion
To use Bonita UI Builder, a Bonita runtime running on your local machine is required (it may be a Bonita Studio or a standalone Bonita runtime).
```

To use the Bonita UI Builder, a Bonita runtime running on your local machine is required (it may be a Bonita Studio or a standalone Bonita runtime).

> Using the provided docker and nginx files above without modification, make sure that the Bonita runtime is accessible at `http://localhost:8080`.

Reviewer: Did we have a way to update or configure this URL ? if yes, it could be nice to add a how to here

You can change the Bonita runtime port by updating the `docker-compose.yml` file and the `nginx.conf` file.

== Authenticate with the UI Builder

> Connect to the  UI Builder with one of the Bonita user credentials.

Reviewer 2: ```suggestion
Log in on Bonita UI Builder with the authentication page of Bonita runtime using the credentials of one of the users of the Bonita runtime database.
```
